src/Forms/DatetimeField.php

```diff
@@ -147,9 +324,11 @@ protected function getISO8601Formatter()
             date_default_timezone_get() // Default to server timezone
         );
         $formatter->setLenient(false);
-        // CLDR iso8601 date.
-        // Note we omit timezone from this format, and we assume server TZ always.
-        $formatter->setPattern('y-MM-dd HH:mm:ss');
+
+        // Note we omit timezone from this format, and we always assume server TZ
+        // ISO8601 date with a standard "T" separator (W3C standard).
+        $formatter->setPattern(DBDatetime::ISO_DATETIME_NORMALISED);
```

Comment:
We should only use ISO_DATETIME_NORMALISED for the localised getFormatter(), and only if html5; Through the rest of framework we assume ISO_DATETIME.